File: postgresql_crud/crud_functions.py
#!/usr/bin/python


def create_todo(conn, title, description):
    """
    Create Todo
    """
    cur = conn.cursor()
    cur.execute("""
    INSERT INTO todos (title, description) VALUES (%s, %s)
    """, (title, description))
    cur.close()
    conn.commit()


def read_todos(conn):
    """ 
    Read Todos
    """
    cur = conn.cursor()
    cur.execute("""
    SELECT * FROM todos
    """)
    todos = cur.fetchall()
    cur.close()
    return todos


def update_todo(conn, id, title, description, completed=False):
    """ 
    Update Todo
    """
    cur = conn.cursor()
    cur.execute("""
    UPDATE todos SET title = %s, description = %s, completed = %s WHERE id = %s
    """, (title, description, completed, id))
    cur.close()
    conn.commit()

def delete_todo(conn, id):
    """ 
    Delete Todo
    """
    cur = conn.cursor()
    cur.execute("""
    DELETE FROM todos WHERE id = %s
    """, (id,))
    cur.close()
    conn.commit()

File: mysql_crud/crud_functions.php
<?php

@include_once ("connect_db.php");

// Create Todo
function createTodo(mysqli $conn, string $title, string $description): int
{
    $sql = 'INSERT INTO todos (title, description) VALUES (?, ?)';
    $stmt = $conn->prepare($sql);
    $stmt->bind_param('ss', $title, $description);
    $stmt->execute();

    return $conn->insert_id;
}

// Read Todos
function readTodos(mysqli $conn): array
{
    $sql = 'SELECT * FROM todos ORDER BY created_at DESC';
    $result = $conn->query($sql);

    $todos = [];
    while ($row = $result->fetch_assoc()) {
        $todos[] = $row;

        echo "<pre>";
        print_r($row);
        echo "</pre>";
    }

    return $todos;
}


// Update Todos
function updateTodo(mysqli $conn, int $id, string $title, string $description, bool $completed = false): void
{
    $sql = "UPDATE `todos` SET `title`='".$title."',`description`='".$description."' WHERE id = " . $id;
    $conn->query($sql);
}

// Delete Todos
function deleteTodo(mysqli $conn, int $id): void
{
    $sql = 'DELETE FROM todos WHERE id = ?';
    $stmt = $conn->prepare($sql);
    $stmt->bind_param('i', $id);
    $stmt->execute();
}